bindata/assets/deployments/downloads-deployment.yaml

```
@@ -123,30 +123,52 @@ spec:
               content = ['<a href="oc-license">license</a>']
               os.symlink('/usr/share/openshift/LICENSE', 'oc-license')
 
-              for arch, operating_system, path in [
-                  ('amd64', 'linux', '/usr/share/openshift/linux_amd64/oc'),
-                  ('amd64', 'mac', '/usr/share/openshift/mac/oc'),
-                  ('amd64', 'windows', '/usr/share/openshift/windows/oc.exe'),
-                  ('arm64', 'linux', '/usr/share/openshift/linux_arm64/oc'),
-                  ('arm64', 'mac', '/usr/share/openshift/mac_arm64/oc'),
-                  ('ppc64le', 'linux', '/usr/share/openshift/linux_ppc64le/oc'),
-                  ('s390x', 'linux', '/usr/share/openshift/linux_s390x/oc'),
-                  ]:
-                basename = os.path.basename(path)
-                target_path = os.path.join(arch, operating_system, basename)
-                os.mkdir(os.path.join(arch, operating_system))
-                os.symlink(path, target_path)
-                base_root, _ = os.path.splitext(basename)
-                archive_path_root = os.path.join(arch, operating_system, base_root)
-                with tarfile.open('{}.tar'.format(archive_path_root), 'w') as tar:
-                  tar.add(path, basename)
-                with zipfile.ZipFile('{}.zip'.format(archive_path_root), 'w') as zip:
-                  zip.write(path, basename)
-                content.append(
-                  '<a href="{0}">oc ({1} {2})</a> (<a href="{3}.tar">tar</a> <a href="{3}.zip">zip</a>)'.format(
-                    target_path, arch, operating_system, archive_path_root
-                  )
-                )
+              ARCHIVE_MAPPINGS = [
+                  ('amd64', 'linux', ['oc', 'oc.rhel8', 'oc.rhel9']),
+                  ('amd64', 'mac', ['oc']),
+                  ('amd64', 'windows', ['oc.exe']),
+                  ('arm64', 'linux', ['oc', 'oc.rhel8', 'oc.rhel9']),
+                  ('arm64', 'mac', ['oc']),
+                  ('ppc64le', 'linux', ['oc', 'oc.rhel8', 'oc.rhel9']),
+                  ('s390x', 'linux', ['oc', 'oc.rhel8', 'oc.rhel9']),
+              ]
+
+              BASE_PATH = '/usr/share/openshift'
+
+              for arch, operating_system, binaries in ARCHIVE_MAPPINGS:
+                  dst_dir = os.path.join(arch, operating_system)
+                  os.makedirs(dst_dir, exist_ok=True)
+
+                  for binary in binaries:
+                      # Construct source path from base + os_arch + binary
+                      os_suffix = f"{operating_system}_{arch}"
```

nit: we could move this `os_suffix` assignment into the scope of the outer cycle.

```suggestion
                  os_suffix = f"{operating_system}_{arch}"
                  for binary in binaries:
                      # Construct source path from base + os_arch + binary
```